pytorch_lightning/callbacks/progress/rich_progress.py

```diff
@@ -324,7 +327,7 @@ def __getstate__(self):
 
     def __setstate__(self, state):
         self.__dict__ = state
-        state["_console"] = Console()
+        self._console = Console(**self._console_kwargs)
```

Comment:
after #10896 `_init_progress` should run in the spawned processes already. We can remove this setstate method entirely (in a new PR) if you agree.

Reply:
:O this would mean we could easily pass in the console object directly through the callback (which would be much nicer imo). @quancs given #10896 has been merged what do you think?

Reply:
How about this
```
console_kwargs: Optional[Dict[str, Any], Console] = None,
```

Reply:
I'll be up for that, but we could add that in a separate PR if you'd prefer :)

Reply:
Yeah. So, let's merge this PR? @awaelchli

Reply:
Yes, are you interested to make a follow up with my proposed change?

Reply:
Yes, I'd like to. Should I propose a issue first?

Reply:
For my proposal here, no, you can just send a PR: https://github.com/PyTorchLightning/pytorch-lightning/pull/10875#discussion_r768676725

I didn't fully understand what @SeanNaren said but yeah, probably fine to send a PR without issue.

Reply:
OK